winml/test/common/protobufHelpers.cpp

```diff
@@ -44,25 +43,31 @@ void FdClose(int fd) {
   }
 }
 
-// Copy and pasted from LOTUS as is.    temporary code to load tensors from protobufs
-bool LoadTensorFromPb(onnx::TensorProto& tensor, std::wstring filePath) {
-  // setup a string converter
-  using convert_type = std::codecvt_utf8<wchar_t>;
-  std::wstring_convert<convert_type, wchar_t> converter;
-
-  // use converter (.to_bytes: wstr->str, .from_bytes: str->wstr)
-  std::string file = converter.to_bytes(filePath.c_str());
-
-  std::ifstream stream(file, std::ios::binary | std::ios::ate);
-  std::streamsize size = stream.tellg();
-  stream.seekg(0, std::ios::beg);
-
-  std::vector<char> buffer(static_cast<size_t>(size));
-  if (stream.read(buffer.data(), size)) {
-    return tensor.ParseFromArray(buffer.data(), static_cast<int>(size));
-  } else {
+// Load Onnx TensorProto from Protobuf File
+bool ProtobufHelpers::LoadOnnxTensorFromProtobufFile(onnx::TensorProto& tensor, std::wstring filePath) {
+  size_t file_size = 0;
+  int block_size = -1;
+  int fd;
+  // Get file descriptor
+  onnxruntime::Status status = onnxruntime::Env::Default().FileOpenRd(filePath, fd);
+  if (!status.IsOK()) {
+    return false;
+  }
+  // Get size of file
+  status = onnxruntime::Env::Default().GetFileLength(fd, file_size);
+  if (status.IsOK()) {
```

Comment:
Don't need be so complex. This is test code, you can directly set the block size 4M. The call of GetFileLength is just to save a few memory when the model size is small, which is not a concern for such a test program. 
It's up to you, you may keep it here.

Reply:
Makes sense, thanks for the suggestion!